### PR TITLE
Adds emp_act() for lockboxes

### PR DESCRIPTION
**Pull request by ghost**
# Document the changes in your pull request

If you will steal a lockbox(for some reason), and would want to open it, you will find an interesting problem - the only ways to open it is to steal a card from someone who has acess to it, or to emag it. This PR adds a way to open this funny boxes with emps.

Basically:
Strong EMP's would break the lockbox(like emags do) and do a small emp on their contents.
Medium EMP's would just break the lockbox as emags do.
Weak EMP's would toggle it open.

# Changelog

Edit the changelog below to reflect the changes made by this PR, even if the changes are minor - required for every PR.
If you add a name after the ':cl', that name will be used in the changelog. Leave it empty to use your GitHub name.

:cl: SuperSlayer
rscadd: EMP's now have an effect on lockboxes
/:cl:

### CODE REVIEW

File: code/game/objects/items/storage/lockbox.dm [hunk 1]
```diff
@@ -76,6 +76,24 @@
...
 	togglelock(user)
 
+/obj/item/storage/lockbox/emp_act(severity)
+	switch(severity)
+		if(1)
+			emag_act(null)
+			for(var/atom/movable/AM in src)
+				AM.emp_act(3)
+		if(2)
+			emag_act(null)
```
Comment (Bibby0110) [inline code/game/objects/items/storage/lockbox.dm]:
```suggestion
			emag_act()
```

File: code/game/objects/items/storage/lockbox.dm [hunk 2]
```diff
@@ -76,6 +76,24 @@
 		return
 	togglelock(user)
 
+/obj/item/storage/lockbox/emp_act(severity)
+	switch(severity)
+		if(1)
+			emag_act(null)
```
Comment (Bibby0110) [inline code/game/objects/items/storage/lockbox.dm]:
```suggestion
			emag_act()
```

File: code/game/objects/items/storage/lockbox.dm [hunk 3]
```diff
@@ -76,6 +76,24 @@
 		return
 	togglelock(user)
 
+/obj/item/storage/lockbox/emp_act(severity)
+	switch(severity)
+		if(1)
```
Comment (Bibby0110) [inline code/game/objects/items/storage/lockbox.dm]: use the defines

**Review by Bibby0110 — CHANGES_REQUESTED** [review]
seems very powerful, EMPs aren't hard to come by

File: code/game/objects/items/storage/lockbox.dm (hunk 3 above)
Comment (ghost) [reply code/game/objects/items/storage/lockbox.dm]: Okay

**Comment by ghost** [comment]
> seems very powerful, EMPs aren't hard to come by

The point is that lockboxes mostly don't contain anything cool enough to be so hard to acess

**Comment by Bibby0110** [comment]
> > seems very powerful, EMPs aren't hard to come by
> 
> The point is that lockboxes mostly don't contain anything cool enough to be so hard to acess

the captains medalbox is a lockbox, and that contains a steal objective

**Comment by ghost** [comment]
Otherway you will have a 6 TC tax on this objective, or you will require to kill a captain with an almost deathsquad armor just to get his card and be able to steal a medal

**Comment by Bibby0110** [comment]
> Otherway you will have a 6 TC tax on this objective, or you will require to kill a captain with an almost deathsquad armor just to get his card and be able to steal a medal

Or just steal the spare ID OR obtain access by tricking the HoP
Or as you say, killing the Captain, which isn’t that hard

**Review by JamieD1 — APPROVED** [review]

**Comment by ghost** [comment]
Now this uses defines + now lockboxes are broken only by heavy EMP's, and light EMP's have only a 60% chance to open the box

**Review by JamieD1 — APPROVED** [review]
Can give it a go

**Pull request merged**